packages/fluentui/react-northstar/src/components/Tree/hooks/flattenTree.ts

```diff
@@ -129,20 +130,21 @@ function flattenSubTree(
         selectedItemIds,
       );
 
-      if (selectedChildrenNum === selectableChildrenNum) {
-        flatTree[id].selected = true;
-        selectedNum++;
-      } else if (selectedChildrenNum > 0) {
-        flatTree[id].selected = 'indeterminate';
-        selectedNum += 0.5; // trick to propagate indeterminate state to ancestors
+      if (selectable) {
+        if (selectedChildrenNum === selectableChildrenNum) {
+          flatTree[id].selected = true;
+          selectedNum++;
+        } else if (selectedChildrenNum > 0) {
+          flatTree[id].selected = 'indeterminate';
+          selectedNum += 0.5; // trick to propagate indeterminate state to ancestors
+        }
       }
-    } else if (selectedItemIds.indexOf(id) >= 0) {
+    } else if (selectable && selectedItemIds.indexOf(id) >= 0) {
```

Comment:
Should we move this to inside the block in the line 147 and so avoid the `selectable` condition in the line 133 and 142?